one_fm/hiring/doctype/electronic_signature_declaration/electronic_signature_declaration.py

```diff
@@ -14,11 +14,18 @@ def after_insert(self):
 		self.update_onboarding_doc()
 		update_onboarding_doc_workflow_sate(self)
 
+	def on_update(self):
+		if self.applicant_signature:
+			self.signature_status = 1
+			self.save(ignore_permissions = True)
+
```

Comment:
It will be a never ending process and what is the purpose of setting the status as 1